### PR TITLE
Add storage policies field for DiskOffering and Primary storage creation Forms

### PR DESCRIPTION
**Pull request by Pearl1594**
Add storage policies field for DiskOffering, COmputeOffering and Primary storage creation Forms

### CODE REVIEW

**Comment by Pearl1594** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@Pearl1594 cc @davidjumani is this ready for testing/merging?

**Comment by Pearl1594** [comment]
@rhtyd this PR is wrt vsphere Advanced capabilities feature. Another form (Compute offering form) needs to be edited as well

**Comment by Pearl1594** [comment]
@blueorangutan package

**Comment by Pearl1594** [comment]
@harikrishna-patnala could you please review the changes made. Thanks

**Comment by rhtyd** [comment]
@davidjumani can you review and advise if this is ready for merging?
@blueorangutan package

File: src/views/infra/AddPrimaryStorage.vue [hunk 1]
```diff
@@ -475,8 +485,22 @@ export default {
...
+          }
+          path += '/' + values.vCenterDataStore
+          url = this.presetupURL('dummy', path)
+        } else if (values.protocol === 'datastorecluster' && values.hypervisor === 'VMware') {
+          path = values.vCenterDataCenter
+          if (path.substring(0, 1) !== '/') {
+            path = '/' + path
+          }
+          path += '/' + values.vCenterDataStore
+          url = this.datastoreclusterURL('dummy', path)
```
[the remark this answers is not in the document]
Comment (Pearl1594) [reply src/views/infra/AddPrimaryStorage.vue]: @harikrishna-patnala I think you looking at an older commit, it's already been addressed in: https://github.com/apache/cloudstack-primate/pull/548/commits/0e86b9a9595281fad39c81c2cdea055c33229e98

File: src/views/infra/AddPrimaryStorage.vue [hunk 2]
```diff
@@ -475,8 +485,22 @@ export default {
...
-        } else if (values.protocol === 'PreSetup') {
+        } else if (values.protocol === 'PreSetup' && values.hypervisor !== 'VMware') {
           url = this.presetupURL(server, path)
+        } else if (values.protocol === 'PreSetup' && values.hypervisor === 'VMware') {
+          path = values.vCenterDataCenter
+          if (path.substring(0, 1) !== '/') {
+            path = '/' + path
+          }
+          path += '/' + values.vCenterDataStore
+          url = this.presetupURL('dummy', path)
```
[the remark this answers is not in the document]
Comment (Pearl1594) [reply src/views/infra/AddPrimaryStorage.vue]: @harikrishna-patnala I think you looking at an older commit, it's already been addressed in: 0e86b9a

[the remark this answers is not in the document]
Comment (harikrishna-patnala) [reply src/views/infra/AddPrimaryStorage.vue]: Yes @Pearl1594, I've checked the recent changes and LGTM

**Comment by Pearl1594** [comment]
@blueorangutan package

**Comment by davidjumani** [comment]
@blueorangutan package

**Pull request merged**